services/api/database/migrations/20241231000100_mysql8_compatibility.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.up = async function(knex) {
+    return knex.schema
+    .alterTable('advanced_task_definition', function (table) {
+        table.specificType('description', 'text').notNullable().alter();
+        table.string('group_name', 300).alter();
+        table.specificType('command', 'text').alter();
+        table.timestamp('deleted').notNullable().alter();
+    })
+    .alterTable('environment', function (table) {
+        table.timestamp('deleted').notNullable().alter();
```

Comment:
Need to update deploy_base_ref and deploy_head_ref here to 250 (from 100)
https://github.com/uselagoon/lagoon/pull/3816/files#diff-94fc09ab7f90affd501ddddf3ae66b4d0b41b4dd22b9b2849f2f692d3b8f3d75R79-R80